src/components/Tables/Table.tsx

```diff
@@ -3,9 +3,15 @@
 import React, { useState, useEffect } from "react";
 import DataTable, { TableColumn } from "react-data-table-component";
 import TableFilterBar, { FilterText } from "@/components/Tables/TableFilterBar";
+import { styled } from "styled-components";
+import { NoSsr } from "@mui/material";
+import SpeechBubbleIcon from "../Icons/SpeechBubbleIcon";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faAnglesUp, faAnglesDown } from "@fortawesome/free-solid-svg-icons";
 
 export interface Datum {
-    [headerKey: string]: string | number | null;
+    data: { [headerKey: string]: string | number | null | undefined };
+    tooltips?: { [headerKey: string]: string | null };
```

Comment:
at what point is it better to structure data as 
```
{ headerkey: {data: xyz, tooltip?: abc} }
```
rather than
```
{ data: {headerKey: xyz}, tooltip?: {headerKey: abc} }
```
?

(I think this is good, just wanted to spawn a conversation about 'higher-level design', especially if it helps the type system!)

Reply:
If designing from scratch, something along the lines of what you're suggesting would be preferable, however this made things easier to integrate with the existing codebase.

Tbh if I was designing it from scratch we would be using arrays and not objects as I feel that makes more sense for something data oriented.